test/api/adapter/test_cognito_adapter.py

```diff
@@ -19,127 +24,232 @@ def setup_method(self):
+        )
+
+    def test_get_resource_server_success(self):
+        expected_response = {
+            "UserPoolId": "user_pool",
+            "Identifier": "identifier",
+            "Name": "name",
+            "Scopes": [
+                {"ScopeName": "scope_name", "ScopeDescription": "scope_description"}
+            ],
+        }
+
+        mock_response = {
+            "ResourceServer": expected_response,
+            "ResponseMetadata": {"HTTPStatusCode": 200},
+        }
+
+        self.cognito_boto_client.describe_resource_server.return_value = mock_response
+
+        actual_response = self.cognito_adapter._get_resource_server(
+            "user_pool", "identifier"
+        )
+
+        self.cognito_boto_client.describe_resource_server.assert_called_once_with(
+            UserPoolId="user_pool", Identifier="identifier"
+        )
+
+        assert actual_response == expected_response
+
+    def test_get_resource_server_fails(self):
+        mock_response = {"ResponseMetadata": {"HTTPStatusCode": 401}}
+        self.cognito_boto_client.describe_resource_server = Mock(
+            return_value=mock_response
+        )
+        with pytest.raises(
+            AWSServiceError,
+            match="The resource server could not be found, please contact system administrator",
+        ):
+            self.cognito_adapter._get_resource_server("pool", "identifier")
+
+    def test_update_resource_server_success(self):
+        new_scope = [{"ScopeName": "new_scope", "ScopeDescription": "new_scope"}]
+
+        self.cognito_boto_client.update_resource_server = Mock()
+
+        mock_describe_response = {
+            "UserPoolId": "user_pool",
+            "Identifier": "identifier",
+            "Name": "name",
+            "Scopes": [
+                {"ScopeName": "existing_scope", "ScopeDescription": "existing_scope"},
+            ],
+        }
+        self.cognito_adapter._get_resource_server = Mock(
+            return_value=mock_describe_response
+        )
+        mock_update_response = {
+            "ResourceServer": {
+                "UserPoolId": "user_pool",
+                "Identifier": "identifier",
+                "Name": "name",
+                "Scopes": [
+                    {
+                        "ScopeName": "existing_scope",
+                        "ScopeDescription": "existing_scope",
+                    },
+                    *new_scope,
+                ],
+            },
+            "ResponseMetadata": {"HTTPStatusCode": 200},
+        }
+        self.cognito_boto_client.update_resource_server = Mock(
+            return_value=mock_update_response
+        )
+        self.cognito_adapter.update_resource_server_scopes(
+            "user_pool", "identifier", new_scope
+        )
+
+        self.cognito_boto_client.update_resource_server.assert_called_once_with(
+            **mock_update_response["ResourceServer"]
+        )
```

Comment:
Should be able to add an assert with the specific output of what the updated response should look like? Or check that the new scope is within `Scopes`?